app/assets/javascripts/initializers/initializeCommentsPage.js.erb

```diff
@@ -272,34 +272,121 @@ function handleKeyUp(event) {
+  var commentCount = user.number_of_comments;
+
+  // var codeOfConduct = user.checked_code_of_conduct;
+  // Code of conduct is false the first time even if the user has accepted it
+  // so we are not using it in the check below
+  // It also seems unnecessary if comment count >= 1 indicates that
+  // it was previously accepted
+
+  // Ensure that user has commented at least once
+  // and we have a non-empty comment to post
+  if (commentCount >= 1 &&
+    event.target.value.trim() !== '') {
+    // get a reference to the submit button for the text area (first level comment vs. reply comment)
+    var parentFormId = event.target.parentElement.parentElement.id || event.target.parentElement.id;
+    var submitButton = document.querySelector('#' + parentFormId + ' input[type="submit"].comment-action-button');
+    submitButton.click();
+  }
+}
+
+// Handler when Ctrl+B/I OR Command+B/I is pressed
+function handleBoldAndItalic(event) {
+  var textArea = event.target;
+
+  var selection = textArea.value.substring(textArea.selectionStart, textArea.selectionEnd);
+  var selectionStart = textArea.selectionStart;
+  var surroundingStr =  event.keyCode === KEY_CODE_B ? "**" : "_";
+
+  replaceSelectedText(textArea, `${surroundingStr}${selection}${surroundingStr}`);
+
+  var selectionStartWithOffset = selectionStart + surroundingStr.length;
+  textArea.setSelectionRange(selectionStartWithOffset, selectionStartWithOffset + selection.length);
+}
+
+// Handler when Ctrl+K OR Command+K is pressed
+function handleLink(event) {
+  var textArea = event.target;
+
+  var selection = textArea.value.substring(textArea.selectionStart, textArea.selectionEnd);
+  var selectionStart = textArea.selectionStart;
+
+  replaceSelectedText(textArea, `[${selection}](url)`);
+
+  // start position + length of selection + [](
+  var startOffset = selectionStart + selection.length + 3;
+
+  // start offset + 'url'.length
+  var endOffset = startOffset + 3;
+
+  textArea.setSelectionRange(startOffset, endOffset);
+}
+
+function replaceSelectedText(textArea, text) {
+  // Chrome and other modern browsers (except FF and IE 8,9,10,11)
+  if(document.execCommand('insertText', false, text)) {
+  }
+  // Firefox (non-standard method)
+  else if(typeof textArea.setRangeText === "function") {
+    textArea.setRangeText(text);
+  }
+
+  /*
+    Disabling IE 8-11 for now as there's no easy way to verify all this
+    We can revisit this if it's really needed and find a jQuery plugin to use
+
+    // IE 8-10
+    else if(document.selection) {
+      const ieRange = document.selection.createRange();
+      ieRange.text = text;
+
+      // Move cursor after the inserted text
+      ieRange.collapse(false); // to the end
+      ieRange.select();
     }
+    else {
+      // Does not support IE 11 yet
+    }
+  */
+}
+
+var KEY_CODE_B = 66;
+var KEY_CODE_I = 73;
+var KEY_CODE_K = 75;
+var ENTER_KEY_CODE = 13;
+
+function handleKeyDown(event) {
+  if (event.ctrlKey || event.metaKey) {
```

Comment:
Hi, could we change this to only be the meta key on a mac? Normally `Control-B` is like the left arrow, and `Control-K` is like `Shift-Right, delete` These keybindings have a very long history, first appearing in... uhm... IDK, Readline or Emacs or something 🤷‍♂️  At this point, they're ubiquitous on a mac, they work in all browsers (eg, in this form I'm currently typing in) and the URL bar. They work in Slack, the dictionary, Spotlight, Atom and in every shell I know of. Node's REPL doesn't use Readline, but it implemented most of the keybindings, because they're so ubiquitous.

They're so ingrained in my head that I accidentally used them 3 times in a row 😝 I wanted to move left over a period and pressed `Control-B`, it inserted 4 asterisks. Surprised, I deleted them, "where was I? Oh yeah" and I promptly pressed `Control-B` again! And, a minute later, one more time.

I think the easy solution is for the modifier to be `Meta` on a Mac (I guess some API binds `Command` to `Meta` instead of giving it its own key?) and the modifier to be `Control` on other systems.

There's also a hard solution of letting users customize their keybindings. I'll admit this appeals to me in general, and especially here, due to our audience. But, not quite enough to champion it 😜

Reply:
Yeah, I agree. I think it should just be `meta` key on Macs/linux and `ctrl` on windows.